model/model_training/custom_datasets/formatting.py

```diff
@@ -25,140 +24,229 @@ def format_system_prefix(prefix, eos_token):
     )
 
 
-class PretrainDatasetEntry(BaseModel):
-    text: str | None = None
+def compute_length(s: str) -> int:
+    return len(re.findall(r"\w+", s)) // 5 + 1
```

Comment:
very arbitrary way of computing length. in the future i would suggest to just quantize to some predefined mesh (e.g. quantize to the nearest multiple of 10) and expose it in the system prompt as `less than 10 words, 10-20 words, ..., 100-200 words, ..., more than 1000 words` or something like that.

Reply:
Yes, length as it is is very experimental. Now that we see it works in principle we can refine it (if we decide to use system-tag profiles in the future). Something similar applies to the numeric values of the properties, which are currently encoded as numbers with one decimal digit, e.g. `quality: 0.4` .. we could reduce the number of tokens by using only a single number, e.g. 0-9.

Reply:
yes, i had the same thought but concluded that single decimal numbers are probably common enough to only require a single token. i do like the semantics of decimal numbers, which probably already are connotated with “strength” or “sliding value” in embedding space.